src/main/drivers/dshot_command.c

```diff
@@ -216,18 +217,29 @@ void dshotCommandWrite(uint8_t index, uint8_t motorCount, uint8_t command, dshot
         for (; repeats; repeats--) {
             delayMicroseconds(DSHOT_COMMAND_DELAY_US);
 
-#ifdef USE_DSHOT_TELEMETRY
-            timeUs_t timeoutUs = micros() + 1000;
-            while (!motorGetVTable().decodeTelemetry() &&
-                   cmpTimeUs(timeoutUs, micros()) > 0);
-#endif
+            // Initialise the output buffers
+            if (vTable.updateInit) {
+                vTable.updateInit();
+            }
+
             for (uint8_t i = 0; i < motorDeviceCount(); i++) {
                 motorDmaOutput_t *const motor = getMotorDmaOutput(i);
                 motor->protocolControl.requestTelemetry = true;
-                motorGetVTable().writeInt(i, (i == index || index == ALL_MOTORS) ? command : DSHOT_CMD_MOTOR_STOP);
+                vTable.writeInt(i, (i == index || index == ALL_MOTORS) ? command : DSHOT_CMD_MOTOR_STOP);
             }
 
-            motorGetVTable().updateComplete();
+            // Don't attempt to write commands to the motors if telemetry is still being received
+            if (vTable.telemetryWait) {
+                (void)vTable.telemetryWait();
+            }
+
+            vTable.updateComplete();
+
+            // Perform the decode of the last data received
+            // New data will be received once the send of motor data, triggered above, completes
+#if defined(USE_DSHOT) && defined(USE_DSHOT_TELEMETRY)
+            vTable.decodeTelemetry();
```

Comment:
```suggestion
            vTable->decodeTelemetry();
```

Reply:
Thanks @KarateBrot. Fixed now.

foo->bar is equivalent to (*foo).bar, i.e. it gets the member called bar from the struct that foo points to.